# Overview

1. Ingestion
2. Data Exploration

# 1. Ingestion

In [1]:
import numpy as np
import pandas as pd

import os

In [2]:
TRAIN_FILE = "/kaggle/working/train.csv"
TEST_FILE = "/kaggle/working/test.csv"

if not os.path.exists(TRAIN_FILE):
    print(f"Unzipping {TRAIN_FILE}...")
    !unzip /kaggle/input/ghouls-goblins-and-ghosts-boo/train.csv.zip

if not os.path.exists(TEST_FILE):
    print(f"Unzipping {TEST_FILE}...")
    !unzip /kaggle/input/ghouls-goblins-and-ghosts-boo/test.csv.zip

Unzipping /kaggle/working/train.csv...
Archive:  /kaggle/input/ghouls-goblins-and-ghosts-boo/train.csv.zip
  inflating: train.csv               
Unzipping /kaggle/working/test.csv...
Archive:  /kaggle/input/ghouls-goblins-and-ghosts-boo/test.csv.zip
  inflating: test.csv                


In [3]:
train_df = pd.read_csv(TRAIN_FILE)

# 2. Data Exploration

In [4]:
train_df.head(10)

id  bone_length  rotting_flesh  hair_length  has_soul  color    type
0   0     0.354512       0.350839     0.465761  0.781142  clear   Ghoul
1   1     0.575560       0.425868     0.531401  0.439899  green  Goblin
2   2     0.467875       0.354330     0.811616  0.791225  black   Ghoul
3   4     0.776652       0.508723     0.636766  0.884464  black   Ghoul
4   5     0.566117       0.875862     0.418594  0.636438  green   Ghost
5   7     0.405680       0.253277     0.441420  0.280324  green  Goblin
6   8     0.399331       0.568952     0.618391  0.467901  white  Goblin
7  11     0.516224       0.536429     0.612776  0.468048  clear   Ghoul
8  12     0.314295       0.671280     0.417267  0.227548   blue   Ghost
9  19     0.280942       0.701457     0.179633  0.141183  white   Ghost

In [5]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371 entries, 0 to 370
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             371 non-null    int64  
 1   bone_length    371 non-null    float64
 2   rotting_flesh  371 non-null    float64
 3   hair_length    371 non-null    float64
 4   has_soul       371 non-null    float64
 5   color          371 non-null    object 
 6   type           371 non-null    object 
dtypes: float64(4), int64(1), object(2)
memory usage: 20.4+ KB


🔑 **Insights**
* Our dataset is quite small with only $371$ records.
* It has mixed features excluding the `id`: $4$ continuous (numerical) and $2$ categorical values